signer/src/storage/postgres.rs

```diff
@@ -403,15 +409,109 @@ impl PgStore {
             "#,
         )
         .bind(chain_tip)
-        .bind(context_window as i32)
-        .bind(txo_type)
+        .bind(min_block_height)
+        .bind(output_type)
         .fetch_optional(&self.0)
         .await
         .map_err(Error::SqlxQuery)?;
 
         Ok(pg_utxo.map(SignerUtxo::from))
     }
 
+    /// Return a block height that is less than or equal to the block that
+    /// confirms the signers' UTXO.
+    ///
+    /// The signers outstanding UTXO is a UTXO that is confirmed on the
+    /// canonical bitcoin blockchain. But regardless of which blockchain
+    /// that is, we know that it is very unlikely to have been affected by
+    /// a reorg of more than 10 blocks deep. That means that if we look back
+    /// at least 10 blocks before the best candidate for the signers' UTXO,
+    /// we are very likely to find a transaction output that is on the best
+    /// bitcoin blockchain.
+    pub async fn minimum_utxo_height(
```

Comment:
I'm wondering if this _pick one, then pick its parent_ is enough: if we have a >10 blocks without activity, then a deposit and two consecutive sweeps (spanning a total of 3 blocks), and those 3 blocks are reorged invalidating the deposit, we will lose track of the original utxo. Is it correct?

About how we may end up in that, we could assume we have the deposit input to be using one of two possible spending path, and in the competing branch (that eventually becomes the canonical one) a tx is spending that same output with another spending path. All of this to say, should we use the buffer `MAX_REORG_BLOCK_COUNT` to constraint `prev_confirmed_height_candidate`?

Reply:
Oh you're right. And I don't think that using the buffer on `prev_confirmed_height_candidate` is enough either. I was really hoping to not go back up the transaction TXO chain but it looks like that is what we need to do.

Basically, we need to start from the UTXO with greatest height and go back at least 10 blocks (which may be more than 10 transactions). It's annoying and might be complex but it's what we should do.

Also, I need to add an additional test for the case that you just spoke about.

Reply:
Oh right, also if we don't actually walk up the transactions chain we could jump from a reorged to a different reorged branch, still missing a canonically confirmed sweep tx.

Reply:
I updated the logic in https://github.com/stacks-network/sbtc/pull/1048/commits/71d852f97bdf9b3da7d24e195c37990287bfb7d0. I'm also going to take another look at the logic (and comments) when I get some fresh eyes. There may be another approach that is simpler.